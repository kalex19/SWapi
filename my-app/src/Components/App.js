import React, { Component } from 'react';
import './App.scss';
// import Header from './Header.js';
// import Controls from './Control.js';
import CardContainer from './CardContainer.js';

export default class App extends Component {
	state = {
		film: {},
		people: [],
		planets: [],
		vehicles: [],
		Favorites: [],
		Error: true,
		Category: null
	};
	//potentially move all state to comp and change app to func comp

	//local storage

	render() {
		return (
			<div className="App">
				{/* {<Header />
				<Controls people={this.people} planets={this.planets} vehicles={this.vehicles} /> */}
				<CardContainer
					film={this.state.film}
					// people={this.people}
					// planets={this.planets}
					// vehicles={this.vehicles}
					// favorites={this.favorites}
					// error={this.error}
				/>
			</div>
		);
	}

	//proptypes
}
