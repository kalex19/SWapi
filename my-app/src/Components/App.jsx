import React, { Component } from 'react';
import './App.scss';
//import Header from './Header';
import Controls from './Controls';
import CardContainer from './CardContainer';

export default class App extends Component {
	state = {
		category: null,
		results: []
	};
	//potentially move all state to comp and change app to func comp

	handleClick = e => {
		this.setState({ category: e.target.name }, () => {
			this.fetchResults();
		});
	};

	fetchResults = () => {
		let results = this.state.category;
		const resultsUrl = `https://swapi.co/api/${results}`;
		fetch(resultsUrl)
			.then(response => response.json())
			.then(results => this.setState({ results: results.results }))
			.catch(error => console.log(error));
	};

	//local storage

	render() {
		return (
			<div className="App">
				{/* <Header /> */}
				<Controls handleClick={this.handleClick} />
				<CardContainer
					results={this.state.results}
					// favorites={this.favorites}
					// error={this.error}
				/>
			</div>
		);
	}

	//proptypes
}
