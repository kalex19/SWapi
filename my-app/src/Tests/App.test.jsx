import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from '../Components/App';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App', () => {
	let wrapper;
	let mockEvent;
	let mockHandleClick;

	beforeEach(() => {
		wrapper = shallow(<App />);
		mockEvent = { preventDefault: jest.fn() };
		mockHandleClick = jest.fn();
	});

	it.only('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<App />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('should mock snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should have default state', () => {
		expect(wrapper.state()).toEqual({
			category: null,
			results: [],
			favortiesCount: 0,
			favorites: []
		});
	});

	it('should call handleClick and set state', async () => {
		wrapper.setState({ category: mockEvent.value });
		await wrapper.instance().handleClick(mockEvent);
		expect(mockHandleClick).toHaveBeenCalled();
	});
});
