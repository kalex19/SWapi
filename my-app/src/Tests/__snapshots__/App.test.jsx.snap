// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should mock snapshot 1`] = `undefined`;
